test/app.test.js

```diff
@@ -0,0 +1,231 @@
+      .then(createdTweet => {
+        return request(app)
+          .delete(`/tweets/${createdTweet._id}`);
+      })
+      .then(res => {
+        expect(res.body).toEqual({ 
+          handle: {
+            handle: 'deleted',
+            email: 'teonna@heintz.com',
+            name: 'teonna',
+            __v: 0,
+            _id: expect.any(String) 
+          },
+          text: 'default text',
+          _id: expect.any(String),
+          __v: 0
+        });
+      });
+  });
+});
+
+
+describe('users', () => {
+
+  beforeEach(done => {
+    return mongoose.connection.dropDatabase(() => {
+      done();
+    });
+  });
+
+  it('creates a user', () => {
+    return request(app)
+      .post('/users')
+      .send({ 
+        handle: 'TA',
+        name: 'Teonna',
+        email: 'teonna@heintz.com'
+      })
+      .then(res => {
+        expect(res.body).toEqual({
+          handle: 'TA',
+          name: 'Teonna',
+          email: 'teonna@heintz.com',
+          _id: expect.any(String),
+          __v: 0
+        });
+      });
+  });
+  it('finds a list of users', () => {
+    return Promise.all(['TT', 'TA'].map(createUser))
+      .then(() => {
+        return request(app)
+          .get('/users');
+      })
+      .then(res => {
+        expect(res.body).toHaveLength(2);
+      });
+  });
+  it('finds by id', () => {
+    return createUser('TMZ')
+      .then(createdUser => {
+        const id = createdUser._id;
+        return request(app)
+          .get(`/users/${id}`)
+          .then(res => {
+            expect(res.body).toEqual({
+              handle: 'TMZ',
+              name: 'teonna',
+              email: 'teonna@heintz.com',
+              _id: expect.any(String),
+              __v: 0
+            });
+          });
+      });
+  });
+
+  it('updates a user by id', () => {
+    return createUser('T_on_A')
+      .then(createdUser => {
+        return request(app)
+          .patch(`/users/${createdUser._id}`)
+          .send({
+            handle: 'T_on_A',
+            name: 'tatiana',
+            email: 'teonna@heintz.com'
+          })
+          .then(() => {
+            return request(app)
+              .get(`/users/${createdUser._id}`)
+              .then(res => {
+                expect(res.body).toEqual({
+                  handle: 'T_on_A',
+                  name: 'tatiana',
+                  email: 'teonna@heintz.com',
+                  _id: expect.any(String),
+                  __v:0
+                });
+              });
+          });
+      });
+  });
+  it('deletes a user by id', () => {
+    return createUser('deleted')
+      .then(createdUser => {
+        return request(app)
+          .delete(`/users/${createdUser._id}`);
+      })
+      .then(res => {
+        expect(res.body).toEqual({
+          handle: 'deleted',
+          name: 'teonna',
+          email: 'teonna@heintz.com',
+          _id: expect.any(String),
+          __v: 0
+        });
+      });
+  });
+
+});
```

Comment:
```suggestion
});

afterAll((done) => {
  mongoose.connection.close(done);
});
```
Using the afterAll hook with done as the passed argument allows mongoose to close the connection at the end of the tests. This resolves an async warning you are currently receiving when running the test suite:

"Jest did not exit one second after the test run has completed. This usually means that there are asynchronous operations that weren't stopped in your tests...